src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -52,7 +52,7 @@ public ModelTypeProvider(InputModelType inputModel, string defaultNamespace, Sou
             _typeFactory = typeFactory!;
             _inputModel = inputModel;
             _sourceInputModel = sourceInputModel;
-            DefaultName = GetValidIdentifier(inputModel.Name); // TODO -- this is only a workaround only to solve the anonymous model names, in other cases, the name is unchanged.
+            DefaultName = inputModel.Name.ToCleanName(); // TODO -- this is only a workaround only to solve the anonymous model names, in other cases, the name is unchanged.
```

Comment:
we could remove the TODO since your change resolves it.

Reply:
removed. @ArcturusZhang